Update Flyway docs to shorten SQL script filenames to prevent conflicts (#10380)
IQSS/dataverse: doc/sphinx-guides/source/developers/sql-upgrade-scripts.rst

We assume you have already read the :doc:`version-control` section and have been keeping your feature branch up to date with the "develop" branch.

> Create a new SQL file in the ``src/main/resources/db/migration`` directory. Name the file to something similar to ``V6.1.0.1.sql``. In this example ``6.1`` represents the current version of Dataverse, with the last digit representing number of the script for that version. Should a newer version be merged while you work on your pull request (PR), you must update your script to a higher number.

Reviewer: ```suggestion
Create a new SQL file in the ``src/main/resources/db/migration`` directory and put a short, meaningful comment at the top. Make the filename something like ``V6.1.0.1.sql``. In this example ``6.1`` represents the current version of Dataverse, with the last digit representing number of the script for that version. (The zero in this example is a placeholder in case the current version has a third number like ``6.1.1``.) Should a newer version be merged while you work on your pull request (PR), you must update your script to the next available number such as ``V6.1.0.2.sql``.
```

> Create a new SQL file in the ``src/main/resources/db/migration`` directory. Name the file to something similar to ``V6.1.0.1.sql``. In this example ``6.1`` represents the current version of Dataverse, with the last digit representing number of the script for that version. Should a newer version be merged while you work on your pull request (PR), you may need to update to a higher number. Generally, PR's merged first take priority, and the order may vary depending on QA considerations at the time.

Reviewer: ```suggestion
Create a new SQL file in the ``src/main/resources/db/migration`` directory. Name the file to something similar to ``V6.1.0.1.sql``. In this example ``6.1`` represents the current version of Dataverse, with the last digit representing number of the script for that version. Should a newer version be merged while you work on your pull request (PR), you must update your script to a higher number.
```

I changed it to "must" RCF style: https://datatracker.ietf.org/doc/html/rfc2119 😄 

The last sentence doesn't make sense to me. 🤔

> 

Reviewer: Can we please add something about the importance of adding a meaningful comment to the .sql file, especially since the filename will no longer indicate what the change is about?

Author: What do you think if I add it at the end of the next paragraph? I think it is better to add that once the reader has the full context of the change.

> Previously, we used the Flyway file naming conventions. However, this approach occasionally led to inadvertent merging of multiple scripts with the same version, such as ``6.0.0.1__0000-wonderful-pr.sql`` and ``6.0.0.1__0001-wonderful-pr.sql``. After careful consideration and analysis, we agreed to adopt the convention mentioned earlier for naming files. This helps us detect conflicts before merging a PR, preventing the development branch from breaking an installation. Additionally, we request that each user writes a meaningful comment on the first line. For more information on Flyway file naming conventions, see https://documentation.red-gate.com/fd/migrations-184127470.html 

Reviewer: ```suggestion
Previously, we used longer, more descriptive file naming conventions supported by Flyway. However, this approach occasionally led to inadvertent merging of multiple scripts with the same version, such as ``V6.0.0.1__0000-wonderful-pr.sql`` and ``V6.0.0.1__0001-lovely-pr.sql`` where ``V6.0.0.1`` must be unique. After careful consideration, we agreed to adopt the convention mentioned above for naming files. This helps us detect conflicts before merging a PR, preventing the develop branch from being undeployable due to a Flyway conflict. For more information on Flyway file naming conventions, see https://documentation.red-gate.com/fd/migrations-184127470.html 
```

I'm trying to communicate that the first part is what matters, the `V6.0.0.1` part can conflict.

Also, I removed the part about the comment from the history lesson and put it higher up under the "do it this way" section. 😄

> Previously, we used the Flyway file naming conventions. However, this approach occasionally led to inadvertent merging of multiple scripts with the same version, such as ``6.0.0.1__0000-wonderful-pr.sql`` and ``6.0.0.1__0001-wonderful-pr.sql``. After careful consideration and analysis, we agreed to adopt the convention mentioned earlier for naming files. This helps us detect conflicts before merging a PR, preventing the development branch from breaking an installation. For more information on Flyway file naming conventions,  see https://documentation.red-gate.com/fd/migrations-184127470.html 

Reviewer: ```suggestion
Previously, we used the Flyway file naming conventions. However, this approach occasionally led to inadvertent merging of multiple scripts with the same version, such as ``6.0.0.1__0000-wonderful-pr.sql`` and ``6.0.0.1__0001-wonderful-pr.sql``. After careful consideration and analysis, we agreed to adopt the convention mentioned earlier for naming files. This helps us detect conflicts before merging a PR, preventing the development branch from breaking an installation. For more information on Flyway file naming conventions, see https://documentation.red-gate.com/fd/migrations-184127470.html 
```
